subprojects/core/src/main/java/org/gradle/internal/buildtree/BuildTreeModelController.java

```diff
@@ -30,11 +29,8 @@ public interface BuildTreeModelController {
      */
     GradleInternal getConfiguredModel();
 
-    ToolingModelScope locateBuilderForDefaultTarget(String modelName, boolean param);
-
-    ToolingModelScope locateBuilderForTarget(BuildState target, String modelName, boolean param);
-
-    ToolingModelScope locateBuilderForTarget(ProjectState target, String modelName, boolean param);
+    @Nullable
+    Object getModel(@Nullable BuildTreeModelTarget target, String modelName, @Nullable Object parameter) throws UnknownModelException;
```

Comment:
🤔 The contract of the method is unclear to me, on one hand it can throw `UnknownModelException`, presumably because the request model was "unknown", on the other it can return `null` and that's a "known" model?

Reply:
The contract of the method is the same as for the existing `ToolingModelScope.getModel`:
- if there is no builder for the model, the exception is thrown
- the builder can return `null` as a result and it is a valid model

I agree that a javadoc is in order in both places